src/backends/tensorrt/plan_backend.h

```diff
@@ -267,9 +300,8 @@ class PlanBackend : public InferenceBackend {
     // The array size is equal to Context::total_bindings_
     std::vector<void*> buffer_bindings_;
 
-    size_t total_batch_size_;
-    std::vector<std::unique_ptr<InferenceRequest>>* requests_;
-    std::shared_ptr<std::vector<std::unique_ptr<InferenceResponse>>> responses_;
+    // The request details of the ongoing model execution
+    std::shared_ptr<Payload> payload_;
```

Comment:
There is only one owner of a payload at any given time, so you should use a unique_ptr here and in the completion_queue_ above. You can then std::move the payload to transfer ownership.